lib/rubocop/cop/lint/duplicate_regexp_character_class_element.rb

```diff
@@ -32,35 +32,53 @@ def on_regexp(node)
           end
         end
 
-        # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         def each_repeated_character_class_element_loc(node)
           node.parsed_tree&.each_expression do |expr|
             next if skip_expression?(expr)
 
             seen = Set.new
-            enum = expr.expressions.to_enum
-            expression_count = expr.expressions.count
+            group_expressions(node, expr.expressions) do |group|
+              group_source = group.map(&:to_s).join
 
-            expression_count.times do |current_number|
-              current_child = enum.next
-              next if within_interpolation?(node, current_child)
+              yield source_range(group) if seen.include?(group_source)
 
-              current_child_source = current_child.to_s
-              next_child = enum.peek if current_number + 1 < expression_count
+              seen << group_source
+            end
+          end
+        end
+
+        private
 
-              if seen.include?(current_child_source)
-                next if start_with_escaped_zero_number?(current_child_source, next_child.to_s)
+        def group_expressions(node, expressions)
+          expressions = expressions.to_a
```

Comment:
This code seems unnecessary, right?

```suggestion
```

Reply:
i was using the array as a "enum" so i can just remove items until it's empty.
it required less state tracking than the original enumerator.

Reply:
part of why i ended up here was because the original method was triggering several linters about method length and complexity, but i couldn't update the counter from a separate method, so in breaking it into smaller methods using a mutable array seemed simplest.